### ISSUE TITLE
(cli): `cdk destroy --force` should use CloudFormation's `FORCE_DELETE_STACK`

### ISSUE THREAD

**Issue by garysassano** [aws/aws-cdk#30344]
### Describe the feature

It would be nice if the CLI command `cdk destroy --force` could take advantage of the new [DeletionMode parameter](https://aws.amazon.com/about-aws/whats-new/2024/05/aws-cloudformation-dev-test-cycle-new-parameter-deletestack-api/) with `FORCE_DELETE_STACK` value to make sure that the stack gets destroyed at all costs, bypassing any possible error that may be encountered.

### Use Case

Make the experience of force destroying a stack less annoying.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.143.0

### Environment details (OS name and version, etc.)

WSL2 Ubuntu 22.04.3 LTS

**Comment by par6n** [comment]
It would be very nice to have CDK leverage this feature. Although, I'm not entirely sure if recycling `--force` flag would be convenient, since the CDK `force` and CloudFormation `FORCE_DELETE_STACK` have two different meanings. Perhaps something like:
```shell
cdk destroy --mode FORCE_DELETE_STACK
```
Which would allow for other modes, if the CF team adds any in the future.

**Comment by nmussy** [comment]
I don't think we should update `destroy --force` to use this new mode, it would be an unexpected behavior for CDK users. The current flag only bypasses CLI confirmation.
I would definitely create a new flag, but a boolean `--force-stack-deletion` might be enough

**Comment by garysassano** [comment]
I agree that it could be problematic to change the behaviour of an existing flag, so I updated the PR.

**Comment by khushail** [comment]
@garysassano , thanks for submitting this FR. I agree with @nmussy ,since we have [CDK Destroy](https://docs.aws.amazon.com/cdk/v2/guide/ref-cli-cmd-destroy.html#ref-cli-cmd-destroy-args), this behavior might be unexpected. Would need core team's input here on the consideration.

**Comment by comcalvi** [comment]
Definitely a new flag on `cdk destroy`, `--force-stack-deletion`
